nmostesting/GenericTest.py

```diff
@@ -420,20 +427,25 @@ def basics(self):
             # Test that the API responds with a 4xx when a missing or invalid token is used
             results.append(self.do_test_authorization(api, "Missing Authorization Header", error_type=None))
             results.append(self.do_test_authorization(api, "Invalid Authorization Token", token=str(uuid.uuid4())))
-            token = self.generate_token([api], True, overrides={"iat": int(time.time() - 7200),
-                                                                "exp": int(time.time() - 3600)})
+            token = None if not self.primary_auth else self.primary_auth.generate_token(
+                [api], True, overrides={"iat": int(time.time() - 7200), "exp": int(time.time() - 3600)})
```

Comment:
The same comment as above regarding idiomatic Python.  This also applies to conditionals in lines 433/434, 437/438 and 443
```suggestion
            token = self.primary_auth.generate_token(
                [api], True, overrides={"iat": int(time.time() - 7200), "exp": int(time.time() - 3600)}) if self.primary_auth else None
```

Reply:
Fixed